packages/lib/models/Setting.ts

```diff
@@ -1384,7 +1384,37 @@ class Setting extends BaseModel {
 				label: () => _('Ignore TLS certificate errors'),
 				storage: SettingStorage.File,
 			},
-
+			'net.proxyEnabled': {
```

Comment:
Hey @laurent22 I seem to be having an issue here, this appears to be reporting `false` no matter whether i tick it or not in the UI. Is this a bug? Other config options seem to work fine but this one

Reply:
It seems to be the case with any boolean config

Reply:
It works fine, but the problem is that the settings aren't loaded by the time you call shimInit(), so you always get the default value. Could you perhaps create a function in the shim called "setupProxySettings()" that would take all the parameters as input, and you would call that after the call from applySettingsSideEffects() in BaseApplication.

You'd need to make sure that this setupProxySettings() function is idempotent because it might be called at any time, whenever settings are changed.

Reply:
Thanks, yeah that's working, ive committed the changes

Reply:
> You'd need to make sure that this setupProxySettings() function is idempotent because it might be called at any time, whenever settings are changed.

do we not want this? As right now the user will have to restart the application if they update the proxy settings.

Reply:
Yes ideally we do, currently all network settings can be changed without restarting the app.

But I don't think restarting the app is required anyway, is it? With the above changed you made in `applySettingsSideEffects`, the proxy configuration should be correctly set every time a setting is changed.

Reply:
> You'd need to make sure that this setupProxySettings() function is idempotent because it might be called at any time, whenever settings are changed.

it sounds like we *don’t* want the function to be idempotent if we want each setting change to have an effect, I.e update the proxy settings without restarting the app. Currently only the first set of changes will have an effect.

Reply:
What I mean is that if you call the function multiples times with the same settings, it should just work (it shouldn't, for example, throw an exception because something has already been initialised).

However we also want the function to run correctly if it's called with different parameters. I see you've prevented that at the moment with the `proxySettings.set` check. Is there any reason for this check, or can it be removed?

Reply:
that set can be removed if that is the case, it sounded before like you didn’t want future changes to have an effect.

Reply:
It doesn’t sound like anything needs to be done here apart from removing the set check

Reply:
Yes sounds like it's al that's needed. There's also the sideEffects update I mentioned before, but hopefully that's straightforward.